pytest_fixtures/component/rh_cloud.py

```diff
@@ -53,35 +43,55 @@ def rhcloud_registered_hosts(organization_ak_setup, mod_content_hosts, rhcloud_s
 
 
 @pytest.fixture
-def rhel_insights_vm(rhcloud_sat_host, organization_ak_setup, rhel_contenthost):
-    """A module-level fixture to create rhel content host registered with insights."""
+def rhel_insights_vm(module_target_sat, organization_ak_setup, rhel_contenthost):
+    """A function-level fixture to create rhel content host registered with insights."""
     # settings.supportability.content_hosts.rhel.versions
     org, ak = organization_ak_setup
-    rhel_contenthost.configure_rex(satellite=rhcloud_sat_host, org=org, register=False)
+    rhel_contenthost.configure_rex(satellite=module_target_sat, org=org, register=False)
     rhel_contenthost.configure_rhai_client(
-        satellite=rhcloud_sat_host,
+        satellite=module_target_sat,
         activation_key=ak.name,
         org=org.label,
         rhel_distro=f"rhel{rhel_contenthost.os_version.major}",
     )
+    # Generate report
+    module_target_sat.generate_inventory_report(org)
+    # Sync inventory status
+    module_target_sat.sync_inventory_status(org)
     yield rhel_contenthost
+    # Delete host
+    host = module_target_sat.api.Host().search(
+        query={"search": f'name={rhel_contenthost.hostname}'}
+    )[0]
+    host.delete()
+    assert not module_target_sat.api.Host().search(query={"search": f'name={host.name}'})
```

Comment:
Why asserting in teardown?

Reply:
good catch, I forgot to move this to test.